index.html

```diff
@@ -202,82 +181,60 @@ <h1>Prototype Viewer MediaWiki 3D GLTF Data, STL data is already available </h1>
+            object.position.z -= center.z;
+
+            // In GLB, Y-Up is required. In STL it's unclear, but likely that
+            // Y-Up is typical, though Z-Up is also common. three.js controls
+            // mostly assume Y-Up. To apply a Z-Up to Y-Up conversion:
+            // object.set( - Math.PI / 2, 0, 0 );
+
+            // Prevent controls from dollying out farther than 10x size.
+            this.controls.maxDistance = size * 10;
+
+            // Constrain camera near and far planes to factors of size. Planes
+            // should fit the content tightly enough that depth buffer
+            // precision is utilized well, but not so tight that the model
+            // clips the near/far planes easily while interacting with controls.
+            this.camera.near = size / 100;
+            this.camera.far = size * 100;
+            this.camera.updateProjectionMatrix();
 
-        TD.center = function (object, modeltype) {
-
-            if (modeltype === 'stl') {
-                if (object.type === 'Group') {
-                    this.center(object.children[0], 'stl');
-                } else if (object.type === 'Mesh') {
-                    object.geometry.center();
-                    object.geometry.computeBoundingSphere();
-
-                    const radius = object.geometry.boundingSphere.radius;
-
-                    // `radius` is the edge of the object's sphere
-                    // We want to position our camera outside of that sphere.
-                    // We'll move `radius` (or more) in all directions (x, y, z), so that we're
-                    // looking at the object somewhat diagonally, which should always show something
-                    // useful (instead of possibly an uninteresting side or top...)
-                    // The exact position of the camera was arbitrarily decided by what looked
-                    // alright-ish for a few files I was testing with.
-                    // sketchfab.com has this at ( 0, -distance, 0 )
-                    // viewstl.com has this at ( 0, 0 distance )
-                    // openjscad.org has this at ( 0, -distance, distance )
-                    // thingiverse.com has this at ( -distance, -distance, distance )
-                    this.camera.position.set(-radius * 1.5, -radius * 1.5, radius);
-                }
-            }
+            // Default viewing angle is arbitrary and subjective, some models
+            // may benefit from a more top-down or front-facing perspective. To
+            // split the difference somewhat, we use a diagonal. Comparisons:
+            // sketchfab.com: ( 0, -distance, 0 )
+            // viewstl.com: ( 0, 0 distance )
+            // openjscad.org: ( 0, -distance, distance )
+            // thingiverse.com: ( -distance, -distance, distance )
+            this.camera.position.x += size * 0.75;
+            this.camera.position.y += size * 0.5;
+            this.camera.position.z += size * 0.75;
```

Comment:
I'm not attached to these weights, but it seemed pretty good in my testing. I think for *most* models you we don't want to be too high, but exceptions (like 3D terrain) do exist, maybe we'd allow the uploader to choose the default viewing angle somewhere down the line later.

Reply:
Looks great!